src/main/java/net/minecraftforge/event/entity/player/PlayerInteractEvent.java

```diff
@@ -208,8 +210,9 @@ public void setCanceled(boolean canceled)
     /**
      * This event is fired on both sides before the player triggers {@link net.minecraft.item.Item#onItemRightClick}.
      * Note that this is NOT fired if the player is targeting a block. For that case, see {@link RightClickBlock}.
-     * Canceling the event clientside causes processing to continue to the other hands,
-     * while canceling serverside will simply do no further processing.
+     *
+     * Let result be the return value of {@link net.minecraft.item.Item#onItemRightClick}, or {@link #cancellationResult} if the event is cancelled.
+     * If we are on the client and result is not {@link EnumActionResult#SUCCESS}, the client will then continue to other hands.
```

Comment:
I think it worth mentionning client does mainhand then, if all else fails, offhand.